src/content/docs/en/reference/api-reference.mdx

```diff
@@ -1048,3 +1048,159 @@ Loads a session by ID. In normal use, a session is loaded automatically from the
   </TabItem>
 
 </Tabs>
+
+### `csp`
+
+
+**Type**: `AstroSharedContextCsp`
+<Since v="6.0.0" />
+
+Astro's CSP runtime APIs enable support for [Content Security Policy (CSP)](https://developer.mozilla.org/en-US/docs/Web/HTTP/Guides/CSP) to help minimize certain types of security threats by controlling which resources a document is allowed to load. This provides additional protection against [cross-site scripting (XSS)](https://developer.mozilla.org/en-US/docs/Glossary/Cross-site_scripting) attacks.
+
+You can customize the `<meta>` element per page from the `Astro` global inside `.astro` components, or the `APIContext` type in endpoints and middleware.
+
+When resources are inserted multiple times or from multiple sources (e.g. defined in your `csp` config and added using [the CSP runtime API](/en/reference/api-reference/#csp)), Astro will merge and deduplicate all resources to create your `<meta>` element.
+
+
+#### `insertDirective`
+
+<p>
+
+**Type:** `(directive: CspDirective) => void`<br />
+<Since v="6.0.0" />
+</p>
+
+Adds a single directive to the current page. You can call this method multiple times to add additional directives.
+
+```astro
```

Comment:
``````suggestion
```astro title="src/pages/index.astro"
``````
Pick anything that makes sense, but since this has to be done at the page level, then it's helpful to emphasize that with a title that shows this is a page in `src/pages/`.